java/util/Vector/Vector.hpp

```diff
@@ -0,0 +1,880 @@
+             * Returns an iterator pointing to the first element in the Vector.
+             *
+             * @return iterator
+             */
+            iterator begin() {
+                return this->original.begin();
+            }
+
+            /**
+             * Returns a const_iterator pointing to the first element in the Vector.
+             *
+             * @return const_iterator
+             */
+            const_iterator begin() const {
+                return this->original.begin();
+            }
+
+            /**
+             * Returns a const_iterator pointing to the first element in the Vector.
+             *
+             * @return const_iterator
+             */
+            const_iterator cbegin() const {
+                return this->original.cbein();
+            }
+
+            /**
+             * Returns an iterator referring to the past-the-end element in the Vector.
+             *
+             * @return iterator
+             */
+            iterator end() {
+                return this->original.end();
+            }
+
+            /**
+             * Returns a const_iterator referring to the past-the-end element in the Vector.
+             *
+             * @return const_iterator
+             */
+            const_iterator end() const {
+                return this->original.end();
+            }
+
+            /**
+             * Returns a const_iterator pointing to the past-the-end element in the Vector.
+             *
+             * @return const_iterator
+             */
+            const_iterator cend() const {
+                return this->original.cend();
+            }
+
+            /**
+             * Array operator.
+             * Uses this operator to access an element at the specified index.
+             *
+             * @param index
+             * @return E
+             */
+            E &operator[](const int index) {
+                if (index < 0 || index >= this->original.size()) {
+                    throw std::invalid_argument("index is out of range");
+                }
+                E *vectorData = this->original.data();
+                return *(vectorData + index);
+            }
+
+            const E &operator[](const int index) const {
+                if (index < 0 || index >= this->original.size()) {
+                    throw std::invalid_argument("index is out of range");
+                }
+                const E *vectorData = this->original.data();
+                return *(vectorData + index);
+            }
+
+            /**
+             * Assignment operator.
+             * Assigns values from right side operands to left side operand.
+             *
+             * @param target
+             * @return Vector<E>
+             */
+            Vector<E> &operator=(const Vector<E> &target) {
+                this->clear();
+                this->ensureCapacity(target.size());
+                const E *targetData = target.original.data();
```

Comment:
I think it should not be changed. Because constructor call an other function (member function) cause a error when build on Travis-CI (GCC C++ compiler).
This operator and that constructor have two tasks different. Assignment operator is used when object is created before.